File: 2244-number-of-laser-beams-in-a-bank/2244-number-of-laser-beams-in-a-bank.cpp
#include <algorithm>

class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int res = 0;
        int prev = 0;

        for (const auto& s: bank) {
            int cur = std::ranges::count(s, '1');
            if (cur > 0) {
                res += cur * prev;
                prev = cur;
            }
        }

        return res;
    }
};

File: 2244-number-of-laser-beams-in-a-bank/2244-number-of-laser-beams-in-a-bank.go
func numberOfBeams(bank []string) int {
    res, prev := 0, 0;
    
    for _, str := range bank {
        cur := 0;
        for _, c := range [] rune (str) {
            if c == '1' {
                cur += 1;
            }
        }
        if cur > 0 {
            res += prev * cur;
            prev = cur;
        }
    }

    return res;
}